velox/exec/Task.cpp

```diff
@@ -22,12 +25,40 @@
 #include "velox/exec/LocalPlanner.h"
 #include "velox/exec/Merge.h"
 #include "velox/exec/PartitionedOutputBufferManager.h"
+#include "velox/exec/Task.h"
 #if CODEGEN_ENABLED == 1
 #include "velox/experimental/codegen/CodegenLogger.h"
 #endif
 
 namespace facebook::velox::exec {
 
+namespace {
+folly::Synchronized<std::vector<std::shared_ptr<TaskListener>>>& listeners() {
+  static folly::Synchronized<std::vector<std::shared_ptr<TaskListener>>>
+      kListeners;
+  return kListeners;
+}
+} // namespace
+
+void registerTaskListener(std::shared_ptr<TaskListener> listener) {
+  listeners().withWLock(
+      [&](auto& listeners) { listeners.push_back(std::move(listener)); });
```

Comment:
should we check if the listener is already added? It could happen due to an incorrect user application.

Reply:
We could have a check and not add a duplicate if already exists.